### PR TITLE
make taylor models an optional dependency

### PR DESCRIPTION
**Pull request by lucaferranti**

### CODE REVIEW

File: docs/Project.toml [hunk 1]
```diff
@@ -3,9 +3,11 @@ Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"
...
 SDPA = "b9a10b5b-afa4-512f-a053-bb3d8080febc"
 SumOfSquares = "4b9e565b-77fc-50a5-a571-1244f986bda1"
+TaylorModels = "314ce334-5f6e-57ae-acf6-00b6e903104a"
 
 [compat]
 Documenter = "0.24"
 IntervalOptimisation = "0.4.1"
 SDPA = "0.2, 0.3, 0.4"
 SumOfSquares = "0.3.6, 0.4, 0.5, 0.6"
+TaylorModels = "0.3, 0.4, 0.5"
```
Comment (schillic) [inline docs/Project.toml:13]:
```suggestion
TaylorModels = "0.3, 0.4, 0.5"

```

File: test/Project.toml [hunk 2]
```diff
@@ -3,10 +3,12 @@ DynamicPolynomials = "7c1d4256-1411-5781-91ec-d7bc3513ac07"
...
 SumOfSquares = "4b9e565b-77fc-50a5-a571-1244f986bda1"
+TaylorModels = "314ce334-5f6e-57ae-acf6-00b6e903104a"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 
 [compat]
 DynamicPolynomials = "0.3, 0.4"
 IntervalOptimisation = "0.4.1"
 SDPA = "0.2, 0.3, 0.4"
 SumOfSquares = "0.3.6, 0.4, 0.5, 0.6"
+TaylorModels = "0.3, 0.4, 0.5"
```
Comment (schillic) [inline test/Project.toml:14]:
```suggestion
TaylorModels = "0.3, 0.4, 0.5"

```

**Review by schillic — COMMENTED** [review]
You need to load `TaylorModels` in `docs/make.jl`.

File: docs/Project.toml (hunk 1 above)
Comment (lucaferranti) [reply docs/Project.toml]: didn't let me commit the suggestion directly, so added it manually (I think), same for below

**Review by schillic — APPROVED** [review]

**Review by mforets — APPROVED** [review]

**Pull request merged**